Add concepts and terminology to getting started guide

dannytsang opened the pull request:
## Proposed change
I have added it to the getting-started pages with the idea of giving people a basic understanding of the features of Home Assistant and describe at a high level what they mean.

This PR has been separated from #23127 as per @c0ffeeca7 feedback.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #22952

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

dannytsang committed:
Created concepts and terminology page.

I have added it to the getting-started pages with the idea of giving people a basic understanding of the features of Home Assistant and describe at a high level what they mean.

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+
+![Dashboard](/images/getting-started/lovelace.png)
+### Integrations
+Software that allows Home Assistant to connect to other software and platforms. For example, Philips Hue integration allow Home Assistant to talk to the Hue Bridge and read data such as the state of your lights through to controlling the lights themselves.
```
Just a few suggestions for clerical changes.

```suggestion
Software that allows Home Assistant to connect to other software and platforms. For example, the Philips Hue integration allows Home Assistant to talk to the Hue Bridge and read data such as the state of your lights through to controlling the lights themselves.
```

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+Once an integration has been added, the hardware and/or data are represented in Home Assistant as [devices and entities](#devices--entities).
+
+### Devices & Entities
+Devices are logical grouping for one or more entities. A device may represent a physical device and the device may have one or more sensors which appear as entities associated to the device. For example, a motion sensor is represented as a device and the motion may provide motion detection, temperature and light levels as entities. Entities have states such as `on` when motion is detected and `of ...
```
```suggestion
Devices are a logical grouping for one or more entities. A device may represent a physical device and the device may have one or more sensors which appear as entities associated to the device. For example, a motion sensor is represented as a device and may provide motion detection, temperature, and light levels as entities. Entities have states such as `on` when motion is detected and `off` when there is no motion.
```

frenck replied:
I would also suggest no to use code-fence `on` and `off` in this case. And just keep it with normal text for this part.

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+![Automations](/images/getting-started/automation-editor.png)
+
+### Scripts
+Similar to automations, repeatable actions that can be run. The difference between scripts and automations is scripts do not have triggers like automations so they cannot automatically run unless it's used in an automation. Scripts are particularly useful if you perform the same actions in different automations or triggering them from a dashboard. More details can be found [here](/integrations/sc ...
```
```suggestion
Similar to automations, repeatable actions that can be run. The difference between scripts and automations is that scripts do not have triggers like automations. This means that scripts cannot automatically run unless they are used in an automation. Scripts are particularly useful if you perform the same actions in different automations or triggering them from a dashboard. More details can be found [here](/integrations/script/).
```

frenck replied:
Additionally: `triggering them from` -> `trigger them from`

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+Scenes are saved states of entities which can be recalled using scripts or automations. For example, a light can be set to a particular brightness, color, etc. More details can be found [here](/integrations/scene/).
+
+### Add-ons
+Depending on your [installation type](/installation), you may have the ability to install add-ons. Add-ons are usually apps that can be run with Home Assistant but provide a quick and easy way to install, configure and run within Home Assistant. Add-ons provide additional functionality where as integrations connects Home Assistant to other apps.
```
```suggestion
Depending on your [installation type](/installation), you may have the ability to install add-ons. Add-ons are usually apps that can be run with Home Assistant but provide a quick and easy way to install, configure, and run within Home Assistant. Add-ons provide additional functionality where as integrations connect Home Assistant to other apps.
```

Maybe it would be worth mentioning that add-ons are third-party software packages, or maybe add a link to the [Glossary](https://www.home-assistant.io/docs/glossary/)? I'm not trying to suggest the text should be copied from the glossary. I personally find it helpful to have different explanations for a concept.

frenck replied:
Interesting, I wonder if we could make a helper or plugin that links to the glossary automatically (or easily) in places. (Not for this PR, just a thought).

frenck replied:
Voila proposal for that: <https://github.com/home-assistant/home-assistant.io/pull/26542>

c0ffeeca7 reviewed (changes requested):
:heart: Love this explainer. I learned from this. Thank you @dannytsang.
But I'm new here, so I'll add @frenck to check my review.

agners commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+Now you're in Home Assistant, let's go over important parts of Home Assistant and what they do.
+
+### Dashboards
+Customizable pages to display information connected to and available in Home Assistant. Dashboards are the first thing you see after the [onboarding process](/getting-started/onboarding). More details can be found [here](/dashboards).
```
A user could be confused if the whole thing is the dashboard, or just certain menu points. How about:

```suggestion
Customizable pages to display information connected to and available in Home Assistant. By default there are two Dashboards, Overview and Energy. The Overview dashboard is the first thing you see after the [onboarding process](/getting-started/onboarding). More details can be found [here](/dashboards).
```

agners commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+Once an integration has been added, the hardware and/or data are represented in Home Assistant as [devices and entities](#devices--entities).
+
+### Devices & Entities
+Devices are logical grouping for one or more entities. A device may represent a physical device and the device may have one or more sensors which appear as entities associated to the device. For example, a motion sensor is represented as a device and the motion may provide motion detection, temperature and light levels as entities. Entities have states such as `on` when motion is detected and `of ...
```
I think we should also emphasize that each entity has one state , and that the state is the main thing we automate on?

frenck replied:
That is not true, attributes are also states.

agners commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+![Home Assistant Device](/images/getting-started/home-assistant-device.png)
+
+### Automations
+A set of repeatable actions that can be set up to run automatically. More details can be found [here](/automation).
```
This is a bit confusing, as "Actions" are one of the three parts what an Automation consists of. Conditions and Actions are the other two. It is a bit a question how deep we go here, but I think we should extend it a bit as this is one of the most important features of Home Assistant in the end.

I think the link should point to `/getting-started/automation`

frenck replied:
It could also help to add a bit of context to it... like short explaining what an action is (For example, turning a light on).

frenck commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+---
+Now you're in Home Assistant, let's go over important parts of Home Assistant and what they do.
+
+### Dashboards
```
One level to much

```suggestion
## Dashboards

```

PS: It is a good practice to add a new line after a header (common Markdown styling).

frenck commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+
+Once an integration has been added, the hardware and/or data are represented in Home Assistant as [devices and entities](#devices--entities).
+
+### Devices & Entities
```
```suggestion
## Devices & Entities

```

frenck commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+
+![Home Assistant Device](/images/getting-started/home-assistant-device.png)
+
+### Automations
```
```suggestion
## Automations

```

frenck commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+
+![Automations](/images/getting-started/automation-editor.png)
+
+### Scripts
```
```suggestion
## Scripts

```

frenck commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+### Scripts
+Similar to automations, repeatable actions that can be run. The difference between scripts and automations is scripts do not have triggers like automations so they cannot automatically run unless it's used in an automation. Scripts are particularly useful if you perform the same actions in different automations or triggering them from a dashboard. More details can be found [here](/integrations/sc ...
+
+### Scenes
```
```suggestion
## Scenes

```

frenck commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+Similar to automations, repeatable actions that can be run. The difference between scripts and automations is scripts do not have triggers like automations so they cannot automatically run unless it's used in an automation. Scripts are particularly useful if you perform the same actions in different automations or triggering them from a dashboard. More details can be found [here](/integrations/sc ...
+
+### Scenes
+Scenes are saved states of entities which can be recalled using scripts or automations. For example, a light can be set to a particular brightness, color, etc. More details can be found [here](/integrations/scene/).
```
```suggestion
Scenes are saved states of entities that can be recalled using scripts or automations. For example, a light can be set to a particular brightness, color, etc. More details can be found [here](/integrations/scene/).
```

frenck replied:
A good bit of context to add here might be: How are those different from automations or scripts?

dannytsang replied:
@frenck I have added the sentence:
_Scenes cannot have conditions are actions like scripts and automations._

is that what you mean?

frenck replied:
Right, yes/no.

So, that makes it sound like scenes are less useful (a more negatively loaded approach, as they can do less than automation and scripts?).

I think the difference between scenes and automation in general, is very unclear in the community. Scenes are often less popular (which is IMHO, not always justified).

Looking at the gist/general question one could have: "Why would I turn on a light using a scene? An automation can do that too?"; which is, of course absolutely true.

(Sorry, thinking out loud here, as I also struggle with a short way to explain it clearly towards an user).

> Scenes are saved states of certain entities that can be recalled using scripts or automation.

Maybe that part is just too technical in this case? "states that can be recalled"?

Or an example could maybe help as well. Like: Set the cosy mood in your living room, by setting all your lights in the mood you are in (brightness, color), setting the thermostat to a nice warm temperature to match the warm lights. By creating a scene for those devices, you now have a cosy scene that you can activate with a single button push that will restore all your device into that cosy state any time in the future... or something better rolling

dannytsang replied:
Gotcha! I must admit I have struggled with the same questions re: scene v.s. automation but I get what you're saying. Let me see what I can conjure up.

frenck commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,43 @@
+### Scenes
+Scenes are saved states of entities which can be recalled using scripts or automations. For example, a light can be set to a particular brightness, color, etc. More details can be found [here](/integrations/scene/).
+
+### Add-ons
```
```suggestion
## Add-ons

```

dannytsang committed:
Incoporated all the text changes from review.

TODO: feedback beyond text changes.

dannytsang committed:
Expanded on automation section.

Tried to describe the differences between a scene v.s a script and automation.

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,55 @@
+
+## Scenes
+
+Scenes are saved states of certain entities that can be recalled using scripts or automations. For example, a light can be set to a particular brightness, color, etc. Scenes cannot have conditions are actions like scripts and automations. More details can be found [here](/integrations/scene/).
```
```suggestion
Scenes are saved states of certain entities that can be recalled using scripts or automations. For example, a light can be set to a particular brightness, color, etc. Scenes cannot have conditions or actions like scripts and automations. More details can be found [here](/integrations/scene/).
```

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,55 @@
+
+## Add-ons
+
+Depending on your [installation type](/installation), you may have the ability to install third party add-ons. Add-ons are usually apps that can be run with Home Assistant but provide a quick and easy way to install, configure, and run within Home Assistant. Add-ons provide additional functionality where as integrations connect Home Assistant to other apps.
```
```suggestion
Depending on your [installation type](/installation), you may have the ability to install third party add-ons. Add-ons are usually apps that can be run with Home Assistant but provide a quick and easy way to install, configure, and run within Home Assistant. Add-ons provide additional functionality whereas integrations connect Home Assistant to other apps.
```

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,55 @@
+
+Devices are a logical grouping for one or more entities. A device may represent a physical device and the device may have one or more sensors which appear as entities associated to the device. For example, a motion sensor is represented as a device and the motion may provide motion detection, temperature and light levels as entities. Entities have states such as on when motion is detected and off ...
+
+Devices and entities can be used in dashboards to [automations](#automations).
```
Not sure I understand this. Do you mean something like _Devices and entities are used in many areas in Home Assistant, e.g. in dashboards and automations_ ?

dannytsang replied:
@c0ffeeca7 good point. I will amend this and expand it a bit.

dannytsang replied:
@c0ffeeca7 how about this:
```
Devices and entities are used throughout Home Assistant. To name a few examples:
* Dashboards can show a state of an entity like if a light bulb is on or off as well as buttons that interact with devices like turning a light bulb on or off.
* An automation can be triggered from a state change on an entity e.g. a light turning on and control another device or entity.
* A pre defined setting for light device saved as a scene.

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,55 @@
+
+## Devices & Entities
+
+Devices are a logical grouping for one or more entities. A device may represent a physical device and the device may have one or more sensors which appear as entities associated to the device. For example, a motion sensor is represented as a device and the motion may provide motion detection, temperature and light levels as entities. Entities have states such as on when motion is detected and off ...
```
```suggestion
Devices are a logical grouping for one or more entities. A device may represent a physical device and the device may have one or more sensors which appear as entities associated to the device. For example, a motion sensor is represented as a device and the motion may provide motion detection, temperature, and light levels as entities. Entities have states such as _on_ when motion is detected and _off_ when there is no motion.
```
Do you think we could maybe add a video illustrating this? Something like
[devices-entities-l.webm](https://user-images.githubusercontent.com/38767475/224471344-14cb0fdc-dc43-4eef-8f04-d1d9441978bb.webm)
I'm not an illustrator. Maybe someone has some suggestions how I could make this prettier.

dannytsang replied:
@c0ffeeca7 I will keep the on and off as plain text as per @frenck previous comment.

As for the video, that would be cool but as you have said I too am not an illustrator and something for the future.

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,55 @@
+## Automations
+
+A set of repeatable actions that can be set up to run automatically. Automations are made of three key components:
+1. Triggers - event(s) that start an automation like when the sun sets.
```
```suggestion
1. Triggers - event(s) that start an automation. For example, when the sun sets.
```
I know I'm being a pain in the ass. But using parallelism can improve readability.

dannytsang replied:
@c0ffeeca7 you're not a PITA and I appreciate the feedback you're doing 👍

dannytsang replied:
@c0ffeeca7 I used *like* instead of for example because I was trying to avoid repetition on point 2 where I use *For example* however, if you think it's better this way then I am happy to amend.

dannytsang committed:
Corrected gramma as per feedback from @c0ffeeca7

dannytsang committed:
Enhanced examples in devices and entities.

Updated scene description to try and differentiate a scene from automation/scripts.

dannytsang committed:
Corrected spelling "everytime".

Britney will not be happy ;-)

c0ffeeca7 commented on source/getting-started/concepts-terminology.markdown:
```diff
@@ -0,0 +1,60 @@
+Devices and entities are used throughout Home Assistant. To name a few examples:
+* [Dashboards](#dashboards) can show a state of an entity like if a light bulb is on or off as well as buttons that interact with devices like turning a light bulb on or off.
+* An [automation](automations) can be triggered from a state change on an entity e.g. a light turning on and control another device or entity.
+* A pre defined setting for light device saved as a [scene](scenes).
```
just one minor typo:
```suggestion
* A predefined setting for light device saved as a [scene](scenes).
```

c0ffeeca7 reviewed (dismissed):
Love the watching movies example to explain what a scene is. :star2:

dannytsang committed:
Update source/getting-started/concepts-terminology.markdown

Fix typo "pre defined".

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

frenck reviewed (approved):
Nice, thanks for bringing this one home @dannytsang ❤️

dannytsang commented:
Lovin' it @frenck 😊 and thanks for your help. Special thanks @c0ffeeca7 and @agners for getting involved.
